### ISSUE TITLE
Preprocess image before processing versions

### ISSUE THREAD

**Issue by jnicklas** [teamcapybara/capybara#59]
Pavel writes on the mailing list:

---

Hi all,

I have this preprocessing and versions in my uploader: http://pastie.org/814075

This used to work in this way:

Original upload has been FIRST:
1) auto oriented
2) all metadata stripped
3) cropped to biggest square possible
4) resized to 400x400

Than this preprocessed 400x400 image was processed by each version so
you got:
321x321 and 90x90 images with rounded corners.

And you got 3 files in this case: 400x400, 321x321, 90x90.

Unfortunately this doesn't work anymore how it behaves is that it runs
the preprocess stuff and saves image 400x400 with original name.

Versions behaves strangely as well:
:full - saves ORIGINAL upload with the name full_original name.
:full/:web - process ORIGINAL upload not the preprocessed one and
saves 321x321 but not autooriented or stripped

and so on.

So now I end up with 400x400, full_ORIGINAL, full_web, thumb_original,
thumb_web...

Basically the preprocess doesn't work at all or I do something wrong.

Any help really appreciated, my processing code: http://pastie.org/814076

Pavel
